File: Entrega-2/src/entregaLabo/ConcreteStrategyInline.java
package entregaLabo;

import robocode.JuniorRobot;

public class ConcreteStrategyInline extends Strategy{

	public ConcreteStrategyInline(JuniorRobot robot) {
		super(robot);
	}
	
	@Override
	public void move() {
		robot.ahead(10);
		robot.turnGunRight(10);
	}
	
	@Override
	public void avoidWall() {
		robot.turnTo(robot.heading+180);
	}
	
	@Override
	public void avoidRobot() {
		 robot.turnRight(90); // Gira rápidamente para cambiar de dirección
		 robot.back(50); // Retrocede para crear distancia
		 robot.turnRight(45); // Gira adicionalmente para seguir en línea
		 robot.ahead(100); // Avanza nuevamente
	}
}

File: Entrega-4/ConcreteStrategyInline.kt
package entregaLabo

import robocode.JuniorRobot

class ConcreteStrategyInline(robot: JuniorRobot) : Strategy(robot) {

    override fun move() {
        robot.ahead(10)
        robot.turnGunRight(10)
    }

    override fun avoidWall() {
        robot.turnTo(robot.heading + 180)
    }

    override fun avoidRobot() {
        robot.turnRight(90)
        robot.back(50)
        robot.turnRight(45)
        robot.ahead(100)
    }
}
